### PR TITLE
Ticket 555

### PR DESCRIPTION
**Pull request by ajj**
Build docs with single command 'python setup.py docs'

### CODE REVIEW

File: docs/sphinx-docs/build_sphinx.py [hunk 1]
```diff
@@ -205,6 +205,7 @@ def retrieve_user_docs():
                     fromhere=os.path.join(SASMODELS_SOURCE_PROLOG,files)
                     tohere=os.path.join(SASMODELS_DEST_PROLOG,files)
                     shutil.copy(fromhere,tohere)
+    else: print "no source directory",SASMODELS_SOURCE_PROLOG, "was found"
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function. Missing a space after the comma and before SASMODELS_SOURCE_PROLOG

Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: the print should probably be on a new line

File: docs/sphinx-docs/build_sphinx.py [hunk 2]
```diff
@@ -216,6 +217,7 @@ def retrieve_user_docs():
                     fromhere=os.path.join(SASMODELS_SOURCE_GPU,files)
                     tohere=os.path.join(SPHINX_SOURCE_USER,files)
                     shutil.copy(fromhere,tohere)
+    else: print "no source directory",SASMODELS_SOURCE_GPU," was found"
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function. spaces after both commas needed.

File: docs/sphinx-docs/build_sphinx.py [hunk 3]
```diff
@@ -227,6 +229,7 @@ def retrieve_user_docs():
                     fromhere=os.path.join(SASMODELS_SOURCE_SESANS,files)
                     tohere=os.path.join(SPHINX_SOURCE_USER,files)
                     shutil.copy(fromhere,tohere)
+    else: print "no source directory",SASMODELS_SOURCE_SESANS," was found"
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function. spaces after both commas needed.

File: docs/sphinx-docs/build_sphinx.py [hunk 4]
```diff
@@ -238,6 +241,7 @@ def retrieve_user_docs():
                     fromhere=os.path.join(SASMODELS_SOURCE_MAGNETISM,files)
                     tohere=os.path.join(SASMODELS_DEST_REF_MODELS,files)
                     shutil.copy(fromhere,tohere)
+    else: print "no source directory",SASMODELS_SOURCE_MAGNETISM," was found"
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function. spaces after both commas needed.

File: docs/sphinx-docs/build_sphinx.py (hunk 2 above)
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: the print should probably be on a new line

File: docs/sphinx-docs/build_sphinx.py (hunk 4 above)
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: the print should probably be on a new line

File: docs/sphinx-docs/build_sphinx.py [hunk 5]
```diff
@@ -275,21 +279,22 @@ def retrieve_user_docs():
...
-            print "Changing toc paths in", files
-            inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
-        if files.startswith("structure"):
-            print "Changing toc paths in", files
-            inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+            # But need to change the path to the model docs in the tocs
+            for files in os.listdir(SASMODELS_DEST_REF_MODELS):
+        #        print files
+                if files.startswith("shape"):
+                    print "Changing toc paths in", files
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function

File: docs/sphinx-docs/build_sphinx.py [hunk 6]
```diff
@@ -275,21 +279,22 @@ def retrieve_user_docs():
...
+                    inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+                if files.startswith("sphere"):
+                    print "Changing toc paths in", files
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function

File: docs/sphinx-docs/build_sphinx.py [hunk 7]
```diff
@@ -275,21 +279,22 @@ def retrieve_user_docs():
...
+                    inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+                if files.startswith("custom"):
+                    print "Changing toc paths in", files
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function

File: docs/sphinx-docs/build_sphinx.py [hunk 8]
```diff
@@ -275,21 +279,22 @@ def retrieve_user_docs():
...
+                    inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+                if files.startswith("structure"):
+                    print "Changing toc paths in", files
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function

File: docs/sphinx-docs/build_sphinx.py [hunk 9]
```diff
@@ -275,21 +279,22 @@ def retrieve_user_docs():
...
-            inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+            # But need to change the path to the model docs in the tocs
+            for files in os.listdir(SASMODELS_DEST_REF_MODELS):
+        #        print files
+                if files.startswith("shape"):
+                    print "Changing toc paths in", files
+                    inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
+                if files.startswith("sphere"):
+                    print "Changing toc paths in", files
+                    inplace_change(os.path.join(SASMODELS_DEST_REF_MODELS,files), "../../model/", "models/")
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: is this line <80 chars long?

File: docs/sphinx-docs/build_sphinx.py [hunk 10]
```diff
@@ -319,6 +324,7 @@ def retrieve_user_docs():
                 fromhere=os.path.join(SASMODELS_DEST_IMG,files)
                 tohere=os.path.join(SASMODELS_DEST_BUILDIMG,files)
                 shutil.copy(fromhere,tohere)
+    else: print "no source directory",SASMODELS_SOURCE_MODELS," was found.\n !!!!NO MODEL DOCS WILL BE BUILT!!!!"
```
Comment (andyfaff) [inline docs/sphinx-docs/build_sphinx.py]: print should be a function. spaces after both commas needed. line needs to be <80 chars. print on new line

File: setup.py [hunk 11]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
         self.cwd = os.getcwd()
 
     def run(self):
+        ''' First builds the sasmodels documentation if the directory
+        is present. Then builds the sasview docs.
+        '''
+        ### AJJ - Add code for building sasmodels docs here:
+        # check for doc path
+        SASMODELS_DOCPATH = os.path.abspath(os.path.join(os.getcwd(),'..','sasmodels','doc'))
+        print "========= check for sasmodels at "+SASMODELS_DOCPATH+"============"
```
Comment (andyfaff) [inline setup.py]: print should be a function. around operators needed.

File: setup.py [hunk 12]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
         self.cwd = os.getcwd()
 
     def run(self):
+        ''' First builds the sasmodels documentation if the directory
+        is present. Then builds the sasview docs.
+        '''
+        ### AJJ - Add code for building sasmodels docs here:
+        # check for doc path
+        SASMODELS_DOCPATH = os.path.abspath(os.path.join(os.getcwd(),'..','sasmodels','doc'))
```
Comment (andyfaff) [inline setup.py]: commas needed after all commas

File: setup.py [hunk 13]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
...
+        print "========= check for sasmodels at "+SASMODELS_DOCPATH+"============"
+        if os.path.exists(SASMODELS_DOCPATH):
+            if os.path.isdir(SASMODELS_DOCPATH):
+                # if available, build sasmodels docs
+                print "============= Building sasmodels model documentation ==============="
```
Comment (andyfaff) [inline setup.py]: print should be a function.

File: setup.py [hunk 14]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
...
+                smdocbuild = subprocess.call(["make","-C",SASMODELS_DOCPATH,"html"])
```
Comment (andyfaff) [inline setup.py]: spaces after all commas

File: setup.py [hunk 15]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
...
+        else:
+            # if not available warning message
+            print " == !!WARNING!! sasmodels directory not found. Cannot build model docs. =="
```
Comment (andyfaff) [inline setup.py]: print should be a function

File: setup.py [hunk 16]
```diff
@@ -151,6 +152,23 @@ def finalize_options(self):
...
+
+        #Now build sasview (+sasmodels) docs
```
Comment (andyfaff) [inline setup.py]: should be a space after the hash

**Comment by andyfaff** [comment]
I wonder if there is anyway we can slowly turn flake8 on, so we get some pep8 checking.

**Comment by ajj** [comment]
We get pylint scores on the builds which is primarily pep8. And the quantified code should be getting things. Problem with that is that it only shows new issues introduced and this file had lots of issues to start with (which I lazily copied)

**Comment by smk78** [comment]
Tested this doc build on my laptop. Seemed ok.

**Pull request merged**